ext/posix/posix.c

```diff
@@ -783,6 +786,9 @@ PHP_FUNCTION(posix_getgrnam)
 
 #if defined(ZTS) && defined(HAVE_GETGRNAM_R) && defined(_SC_GETGR_R_SIZE_MAX)
 	buflen = sysconf(_SC_GETGR_R_SIZE_MAX);
+#if defined(__FreeBSD__) && defined(_SC_PAGESIZE)
```

Comment:
if it's not supported on all FreeBSD, why do you still try to use `_SC_GETGR_R_SIZE_MAX` and then just fall back?

Reply:
It's hardcoded as `-1`.
https://cgit.freebsd.org/src/tree/lib/libc/gen/sysconf.c#n363

Reply:
I think what bukka meant is why bothering with _SC_GETGR_R_SIZE_MAX if it returns -1 anyway (means no hard limit on the getgrnam_r buffer in theory).

Reply:
So I looked a bit around, also dragonflybsd and linux/musl returns -1 too, so would checking -1 and no errno set do the trick ?

Reply:
Give me some times to recheck. At least in one circumstances with FreeBSD 12 I can see that:
```<?php var_dump(posix_getpwuid(0)); ?>``` just returns ```bool(false)``` where not on recent FreeBSD versions. I will need to recheck the codebase updates.

Reply:
So it looks like that when ZTS is enabled this issue occurs returns ```bool(false)```. And when ZTS is not enabled this issue does not occur. So setting no errno should resolve the problem.

Reply:
Closing this in lieu of https://github.com/php/php-src/pull/13187